game-core/src/system/camera.rs

```diff
@@ -0,0 +1,48 @@
+use amethyst::renderer::Camera; 
+use amethyst::{
+    core::Transform,
+    ecs::{Join, ReadStorage, System, WriteStorage},
+};
+
+use crate::component::Player;
+
+pub struct Movement;
+
+impl<'s> System<'s> for Movement {
+    type SystemData = (
+        ReadStorage<'s, Player>,
+        ReadStorage<'s, Camera>,
+        WriteStorage<'s, Transform>,
+    );
+
+    fn run(&mut self, (players, cameras, mut transforms): Self::SystemData) {
+        let mut player_x: f32 = 0.0;
+        let mut player_y: f32 = 0.0;
+
+        // let mut camera_x: f32 = 0.0;
+        // let mut camera_y: f32 = 0.0;
+
+        // let mut camera_safe_x: f32 = 0.0;
+        // let mut camera_safe_y: f32 = 0.0;
+
+        for (_, transform) in (&players, &mut transforms).join() {
+            player_x = transform.translation.x - 256.0 as f32;// * 5.0;
+            player_y = transform.translation.y - 256.0 as f32;// * 5.0;
+        }
+
+        for (_, transform) in (&cameras, &mut transforms).join() {
+            let camera_x = transform.translation.x as f32;// * 5.0;
+            let camera_y = transform.translation.y as f32;// * 5.0;
+            let camera_safe_x = transform.scale.x/4 as f32;
+            let camera_safe_y = transform.scale.y/4 as f32;
+
+            if (player_x - camera_x).abs() > camera_safe_x || (player_y - camera_y).abs() > camera_safe_y {
```

Comment:
you should be able to do the math here on the vecs themselves rather than the values inside. Something like

`(player.transform.translation.truncate() - camera...).magnitude2() > camera_safe_x.pow(2)`

Check out cgmath::InnerSpace documentation, thats where they hid all the normal euclidean mathy stuff we want.